src/huggingface_hub/utils/tags.py

```diff
@@ -0,0 +1,140 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" Tagging utilities. """
+
+
+class AttributeDictionary(dict):
```

Comment:
I wonder if instead of having an `dict` subclass we could solve it with a data class that has dynamic fields (see [make_dataclass](https://docs.python.org/3/library/dataclasses.html#dataclasses.make_dataclass)). Is this an option that was considered?

Reply:
It was in Lysandre's previous work on this, I wasn't a fan of how many subclasses and dataclasses this wound up having in that version personally.

I'm also not too familiar with dataclasses (I've done no work with them), so I'll need to look into it :smile:

Reply:
I think the issue with a data class is there is something really important inheriting from a dict does for us:

If it starts with a digit, all we have to do is tell the user to look at it as a dictionary rather than an attribute, and report that it exists in the dictionary through the `repr`.

Let me know if the dataclass would succeed in something similar, but that's the main extreme case that this solves.

Reply:
If we used a namedclass I'm pretty sure we could easily add `repr` access as well by an override that access the attribute.

But the main question here is how to easily generate a dynamic namedclass from the API response, and it seems it would be more complicated, so I think we can move ahead with your solution!